docs/rules/no-node-access.md

```diff
@@ -0,0 +1,42 @@
+# Disallow direct Node access (no-node-access)
+
+The Testing Library already provides methods for querying DOM elements.
+
+## Rule Details
+
+This rule aims to disallow DOM traversal using native HTML methods and properties, such as `closest`, `lastChild` and all that returns another Node element from an HTML tree.
+
+Examples of **incorrect** code for this rule:
+
+```js
+screen.getByText('Submit').closest('button'); // chaining with Testing Library methods
+```
+
+```js
+const buttons = screen.getAllByRole('button');
+expect(buttons[1].lastChild).toBeInTheDocument();
+```
+
+```js
+const buttonText = screen.getByText('Submit');
+const button = buttonText.closest('button');
+```
+
+```js
+document.getElementById('submit-btn').closest('button');
+```
+
+Examples of **correct** code for this rule:
+
+```js
+const button = screen.getByRole('button');
+expect(button).toHaveTextContent('submit');
```

Comment:
Last example requirement: I think it would be nice to add a more elaborated example using `within` which is a nice way of finding elements from another one.

Reply:
> Are we waiting then for the checks you mentioned in your last comments

After including this suggestion of using `within` I think we're good to go.

Reply:
Done! I've also updated the examples and testing cases. I've added a verification on the import declarations to check whether a testing-library package is being imported before triggering an error message.